From 46ece5d3ce2e2ae96e0f5298d164e8d73f97b247 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Tue, 09 Mar 2021 06:23:13 +0200
Subject: [PATCH] [SW-0]: greco increase stack size to 4k from 512 bytes

with recent code changes in preboot, existing stack size of 512 bytes
is not sufficient. Stack overflow is seen and corrupts global variables
like hl_ts_data. Increasing stack size to 1k resolves the problem.
But keeping stack size to 4K to align with Gaudi and Greco secure boot
flow stack size

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I66e6326fdaae450fa001c6254fff5eaadcc0686e
---

diff --git a/goya2/preboot.lds.S b/goya2/preboot.lds.S
index 0c0b725..0e4134f 100644
--- a/goya2/preboot.lds.S
+++ b/goya2/preboot.lds.S
@@ -30,7 +30,7 @@
 	.stack : {
 		. = ALIGN(8);
 		__stack = . ;
-		. = . + 512 ;
+		. = . + 4096 ;
 		__stack_top = . ;
 	} > .sram
 
diff --git a/greco/preboot.lds.S b/greco/preboot.lds.S
index 0c0b725..0e4134f 100644
--- a/greco/preboot.lds.S
+++ b/greco/preboot.lds.S
@@ -30,7 +30,7 @@
 	.stack : {
 		. = ALIGN(8);
 		__stack = . ;
-		. = . + 512 ;
+		. = . + 4096 ;
 		__stack_top = . ;
 	} > .sram
 
